app/workers/machine_translation_worker.rb

```diff
@@ -0,0 +1,41 @@
+class MachineTranslationWorker
+  include Sidekiq::Worker
+
+  def perform(target, author)
+    target = YAML::load(target)
+    author = YAML::load(author)
+    field = DynamicAnnotation::Field.joins(:annotation).where('annotations.annotation_type' => 'language', 'annotations.annotated_type' => target.class.name, 'annotations.annotated_id' => target.id.to_s, field_type: 'language').first
+    src_lang = field.value unless field.nil?
+    mt = target.annotations.where(annotation_type: 'mt').last
+    unless mt.nil? or src_lang.nil?
+      translations = []
+      text = target.text
+      # Assume languages format is ['en', 'ar', ...]
+      languages = target.project.settings[:languages] unless target.project.settings.nil?
+      languages = languages - [src_lang] unless languages.nil?
+      languages.each do |lang|
+        begin
+          response = AlegreClient::Request.get_mt(CONFIG['alegre_host'], { text: text, from: src_lang, to: lang }, CONFIG['alegre_token'])
+          mt_text = response['type'] == 'mt' ? response['data'] : nil
+        rescue
+          mt_text = nil
+        end
+        unless mt_text.nil?
+          lang_name = TwitterCldr::Shared::LanguageCodes.to_language(lang, :iso_639_1)
```

Comment:
We don't need to store the language name in the annotation, since we don't know in advance the locale which should show the name of the target language.